modules/lib/core/index.d.ts

```diff
@@ -142,6 +142,318 @@ export interface ScriptValue {
 export type XpRequire = <Key extends keyof XpLibraries | string = string>(path: Key) =>
     Key extends keyof XpLibraries ? XpLibraries[Key] : unknown;
 
+export type LiteralUnion<T extends U, U = string> = T | (U & Record<never, never>);
+
+export type StrictMergeInterfaces<
+    A,
+    B = Record<string, never>
+> = B extends Record<string, never>
+    ? A
+    : Pick<A, Exclude<keyof A, keyof B>> & B;
+
+export interface ComplexCookie {
+    /**
+     * The value of the cookie (optional).
+     * 
+     * @type string
+     */
+    value: string
+    /**
+     * A comment (rfc2109) to document the cookie (optional).
+     * 
+     * @type string
+     */
+    comment?: string
+    /**
+     * The expiration date and time for the cookie (optional).
+     *
+     * @type string
+     */
+    expires?: Date;
+    /**
+     * The domain name for which the cookie is set.
+     *
+     * @type string
+     */
+    domain?: string
+    /**
+     *  Indicates whether the cookie should not be accessible via JavaScript (optional).
+     *
+     * @type string
+     */
+    httpOnly?: boolean
+    /**
+     * The maximum age of the cookie in seconds (optional).
+     *
+     * @type string
+     */
+    maxAge?: number
+    /**
+     * The path on the server where the cookie should be available.
+     *
+     * @type string
+     */
+    path?: string
+    /**
+     *  Specifies the SameSite attribute (draft RFC) for the cookie (optional).
+     *
+     * @type string
+     */
+    sameSite?: LiteralUnion<'lax' | 'strict' | 'none'>;
+    /**
+     * Indicates whether the cookie should only be sent over HTTPS (optional).
+     *
+     * @type string
+     */
+    secure?: boolean
+}
+
+export type RequestBranch = 'draft' | 'master';
+export type RequestGetHeaderFunction = (headerName: string) => string | null;
+export type RequestMethod = 'GET' | 'POST' | ' HEAD' | 'OPTIONS' | ' PUT' | 'DELETE' | ' TRACE' | 'CONNECT' | ' PATCH' | 'PROPFIND' | ' PROPPATCH' | 'MKCOL' | ' COPY' | 'MOVE' | ' LOCK' | 'UNLOCK';
+export type RequestMode = 'edit' | 'inline' | 'live' | 'preview' | 'admin';
+export type RequestParams = Record<string, string | string[]>;
+export type RequestScheme = 'http' | 'https';
```

Comment:
how about ws/wss ? not sure they are ever visible - should be tested.

Reply:
ws/wss are allowed, just not autocompleted

Reply:
I tested the chat in app features, the websocket request doesn't go to the "get" method, but directly to the "webSocketEvent" method, which doesn't receive the request object, but an event object.

Reply:
The first request has webSocket: false, the second has webSocket: true, but is still has scheme: http, the third and ongoing are just a events, not passing through the get function.

Reply:
Added webSocketEvent

Reply:
Now, more like WebSocketEventMapper